gulpfile.js

```diff
@@ -22,6 +23,115 @@ const rollup = require('rollup');
 const rollupString = require('rollup-plugin-string');
 const rollupTypescript = require('rollup-plugin-typescript2');
 
+const rxjsTypes = new Set(['Observer']);
+
+const transformRxjsImportText = (importName) =>
+  importName === 'EMPTY'
+    ? 'rxjs/internal/observable/empty'
+    : rxjsTypes.has(importName)
+    ? 'rxjs/internal/types'
+    : importName[0].toLowerCase() === importName[0]
+    ? `rxjs/internal/observable/${importName}`
+    : `rxjs/internal/${importName}`;
+
+const transformRxjsOperatorsImportText = (importName) => `rxjs/internal/operators/${importName}`;
+
+const transformImport = (name, callback, statements) => {
+  const idx = statements.findIndex(
+    (statement) => typescript.isImportDeclaration(statement) && statement.moduleSpecifier.text === name,
+  );
+  if (idx !== -1) {
+    const beforeIdx = statements.slice(0, idx);
+    const foundImport = statements[idx];
+    const afterIdx = statements.slice(idx + 1);
+
+    const newImports = foundImport.importClause.namedBindings.elements.map((elem) =>
+      typescript.createImportDeclaration(
+        foundImport.decorators,
+        foundImport.modifiers,
+        typescript.createImportClause(
+          undefined,
+          typescript.createNamedImports([typescript.createImportSpecifier(elem.propertyName, elem.name)]),
+        ),
+        typescript.createStringLiteral(
+          callback(elem.propertyName === undefined ? elem.name.getText() : elem.propertyName.getText()),
+        ),
+      ),
+    );
+
+    return beforeIdx.concat(newImports, afterIdx);
+  }
+
+  return statements;
+};
+
+const createCustomTransform = (context) => {
+  return (node) => {
+    const visitor = (node) => {
+      if (typescript.isSourceFile(node)) {
+        let statements = node.statements;
+        statements = transformImport(`rxjs`, transformRxjsImportText, statements);
```

Comment:
The way you're doing this is `O(n*m)` complexity, where `n` is the number of statements in the file and `m` is the number of imports to map. Please change it to `O(n)`. I'll leave it as an exercise to figure out how :)

Note, I fully realize it's probably not an issue here, but it's good practice to always use the most optimal solution.